src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/resolver/CoreScalarResolverTest.kt

```diff
@@ -0,0 +1,16 @@
+package it.krzeminski.snakeyaml.engine.kmp.resolver
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.regex.shouldMatch
+import io.kotest.matchers.regex.shouldNotMatch
+import io.kotest.matchers.shouldBe
+import it.krzeminski.snakeyaml.engine.kmp.nodes.Tag
+
+class CoreScalarResolverTest: FunSpec({
+    test("resolve implicit integer") {
+        CoreScalarResolver.INT shouldMatch "0o1010"
```

Comment:
Failing for JS:
`AssertionError: AssertionError: Regex '/^([-+]?[0-9]+)|(0o[0-7]+)|(0x[0-9a-fA-F]+)$/gu' should match 0o1010`

Reply:
I'm playing on https://regex101.com/ and I'm having problems matching this regex for this string even for the Java flavor.

Reply:
BTW, I recommend https://www.debuggex.com/ that helped me understand what's wrong.

Reply:
Can you explain, please? 
Because that does not really make sense.
The current regex should match fine. 
And (unless I miscopied on mobile) debuggex is also happy with the current state. 

The change you propose does not change any functionality as long as whole strings are matched and even then the test would not fail, besides that the change is part of my readability PR, just with non-capturing groups.

Reply:
See the description in https://github.com/krzema12/snakeyaml-engine-kmp/pull/403